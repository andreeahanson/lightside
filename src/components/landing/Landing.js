import React from 'react';
import Crawl from 'react-star-wars-crawl';
import './Landing.css';

const Landing = ({ films }) => {
  const movie = films[Math.floor(Math.random() * films.length)] || {};
  return (
    <Crawl>
      <h3>{movie.title}</h3>
      <p>{movie.opening_crawl}</p>
      <h4>{movie.release_date}</h4>
    </Crawl>
  );
};

export default Landing;
