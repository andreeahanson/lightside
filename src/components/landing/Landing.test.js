import React from 'react';
import { shallow } from 'enzyme';
import Landing from './Landing';
import sampleFilms from '../../sampleDataDirectory/sampleFilms';

describe('Landing', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Landing films={sampleFilms.results} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
