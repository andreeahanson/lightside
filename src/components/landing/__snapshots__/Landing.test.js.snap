// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should match the snapshot 1`] = `ShallowWrapper {}`;
